mappings/loc.py

```diff
@@ -8,32 +10,111 @@ def __init__(self, source):
+            'is_part_of': ('partof', '{0}|collection'),
+            'abstract': 
                 ('description', '{0}')
-            ],
-            'subjects': [('subjects', '{0}||')],
-            'rights': (['rights_advisory', 'copyra', 'copyri'], 'met|{0}|{1}|{2}|'),
-            'has_part': [('pdf', '1|{0}|loc|application/pdf|{{"catalog": false, "download": true, "reader": false, "embed": false}}')]
+            ,
         }
 
     def applyFormatting(self):
+        self.record.has_part = []
         self.record.source = 'loc'
-        self.record.source_id = self.record.identifiers[0]
         self.record.medium = self.record.medium[0]
-        self.record.spatial = self.record.spatial.split(':')[0].strip(' ')
+
+        #Convert string repr of list to actual list
+        itemList = ast.literal_eval(self.record.identifiers[1])
+
+        self.record.identifiers[0], self.record.identifiers[1], self.record.source_id = self.formatIdentifierSourceID(itemList)
+
+        self.record.publisher, self.record.spatial = self.formatPubSpatial(itemList)
+
+        self.record.extent = self.formatExtent(itemList)
+
+        self.record.subjects = self.formatSubjects(itemList)
+
+        self.record.rights = self.formatRights(itemList)
+
+        self.record.languages = self.formatLanguages(itemList)
+
+    #Identifier/SourceID Formatting
+    def formatIdentifierSourceID(self, itemList):
+        newIdentifier = itemList
+        newIdentifier['call_number'][0] = f'{newIdentifier["call_number"][0]}|call_number'
+        lccnNumber = self.record.identifiers[0][0]  #lccnNumber comes in as an array and we need the string inside the array
+        callNumber = newIdentifier['call_number'][0].strip(' ')
+        sourceID = lccnNumber
+        return (lccnNumber, callNumber, sourceID)
+    
+    #Publisher/Spatial Formatting
+    def formatPubSpatial(self, itemList):
+        pubArray = []
+        spatialArray = []
+        for elem in itemList['created_published']:
+            if ':' not in elem:
+                createdPublishedList = elem.split(',', 1)
+                pubLocation = createdPublishedList[0].strip(' ')
+                if ',' in createdPublishedList[1]:
+                    pubOnly = createdPublishedList[1].split(',')[0].strip(' ')
+                    pubArray.append(pubOnly)
+                spatialArray.append(pubLocation)
+            else:
+                pubLocatAndPubInfo = elem.split(':', 1)
+                pubLocation = pubLocatAndPubInfo[0].strip()
+                pubInfo = pubLocatAndPubInfo[1]
+                pubOnly = pubInfo.split(',', 1)[0].strip()
+                pubArray.append(pubOnly)
+                spatialArray.append(pubLocation)
+        return (pubArray, spatialArray)
+    
+    #Extent Formatting
+    def formatExtent(self, itemList):
+        extentArray = []
+
+        if 'medium' in itemList:
+            for elem in itemList['medium']:
+                extentArray.append(elem)
```

Comment:
seems like you're just returning `itemList['medium']` here right? Do you need the loop at all?

Reply:
You're right. I'll just use the extend method for the extentArray instead to get all the elements of the medium array.